src/harness/virtualFileSystemWithWatch.ts

```diff
@@ -52,80 +53,61 @@ interface Array<T> { length: number; [n: number]: T; }`
         return host;
     }
 
-    export interface File {
-        path: string;
-        content: string;
-        fileSize?: number;
-    }
-
-    export interface Folder {
-        path: string;
-    }
-
-    export interface SymLink {
-        /** Location of the symlink. */
-        path: string;
-        /** Relative path to the real file. */
-        symLink: string;
-    }
-
-    export type FileOrFolderOrSymLink = File | Folder | SymLink;
-    export function isFile(fileOrFolderOrSymLink: FileOrFolderOrSymLink): fileOrFolderOrSymLink is File {
-        return isString((fileOrFolderOrSymLink as File).content);
-    }
-
-    export function isSymLink(fileOrFolderOrSymLink: FileOrFolderOrSymLink): fileOrFolderOrSymLink is SymLink {
-        return isString((fileOrFolderOrSymLink as SymLink).symLink);
-    }
-
-    interface FSEntryBase {
-        path: Path;
-        fullPath: string;
-        modifiedTime: Date;
-    }
-
-    interface FsFile extends FSEntryBase {
-        content: string;
-        fileSize?: number;
+    export function createVirtualServerHost(params: VirtualServerHostCreationParameters): VirtualServerHost {
+        const host = new VirtualServerHost(params);
+        host.init();

```

Comment:
Isnt constructor already doing init ?

Reply:
TestServerHost needs to construct itself differently from VirtualServerHost. This is a clunky attempt to keep the common work in VirtualServerHost's constructor and put the distinct work in TestServerHost.construct, TestServerHost.init and VirtualServerHost.init. I extracted the VirtualServerHost initialisation to a new subclass, which puts the distinct work in the subclass constructors and the common work in the base constructor.